### PR TITLE
Confluent Cloud Integration

### PR DESCRIPTION
**Pull request by dmccoystephenson**
# Purpose
These changes allow the S3D to communicate with Confluent Cloud.

## Notes
-  Some new launch configurations have been introduced for local testing.
- These changes have been tested with both a local instance of kafka as well as an instance of kafka hosted by Confluent Cloud.
- It has been verified that we can push data to an S3 bucket with these changes.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "java",
+            "name": "S3D Local",
+            "request": "launch",
+            "mainClass": "us.dot.its.jpo.ode.aws.depositor.AwsDepositor",
+            "projectName": "jpo-aws-depositor",
+            "args": ["--bootstrap-server", "<kafka host>:9092",
```
Comment (payneBrandon) [inline .vscode/launch.json:13]: Are these "< >" values just stand-ins to be replaced by the developer?

File: pom.xml [hunk 2]
```diff
@@ -59,6 +59,20 @@
       <artifactId>json</artifactId>
       <version>20200518</version>
     </dependency>
+
+    <dependency>
```
Comment (payneBrandon) [inline pom.xml:63]: Are we using slf4j for logging?

File: run.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# note: must be run from the project root folder
```
Comment (payneBrandon) [inline run.sh:1]: If we're going to add a script to test running things, let's add a write-up about it in the README.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+version: '2'
```
Comment (payneBrandon) [inline docker-compose.yml:1]: Let's add some info about this docker-compose and the docker-compose-confluent-cloud to the README so we know what it's used for

File: src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java [hunk 5]
```diff
@@ -138,17 +142,24 @@ public static void main(String[] args) throws Exception {
 			AWS_SESSION_TOKEN = profile.get(K_AWS_SESSION_TOKEN).toString();
 			AWS_EXPIRATION = profile.get(K_AWS_EXPIRATION).toString().split("\\+")[0];
 		}
-		
 
 		// Properties for the kafka topic
 		Properties props = new Properties();
 		props.put("bootstrap.servers", endpoint);
+
+		props.put("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
```
Comment (payneBrandon) [inline src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java:150]: Are these just needed with the new Kafka version?

Comment (dmccoystephenson) [reply src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java:150]: These already existed, I've just moved them a bit higher in the file to match the order of property declarations in other projects.

File: docker-compose.yml (hunk 4 above)
Comment (dmccoystephenson) [reply docker-compose.yml:1]: A section on docker-compose files has been added to the README.

File: run.sh (hunk 3 above)
Comment (dmccoystephenson) [reply run.sh:1]: A section on this script has been added to the README.

Comment (payneBrandon) [reply run.sh:1]: Let's expand that section a bit to include that we're expecting the jar to already exist or run a build in the script before executing the jar. That 2nd option may be the better one to ensure we have a fresh build before execution.

File: pom.xml (hunk 2 above)
Comment (payneBrandon) [reply pom.xml:63]: Looks like we use slf4j in the AwsDepositor file, must have been left out somehow...strange.

File: run.sh (hunk 3 above)
Comment (dmccoystephenson) [reply run.sh:1]: Good call. The run.sh script has been altered to build the project before running it.

File: src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java [hunk 6]
```diff
@@ -106,7 +110,7 @@ public static void main(String[] args) throws Exception {
 
 		// S3 properties
 		bucketName = cmd.getOptionValue("bucket-name");
-		awsRegion = cmd.getOptionValue("region", "us-east-1");
+		awsRegion = cmd.getOptionValue("region", "us-east-2");
```
Comment (payneBrandon) [inline src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java:113]: I'm not sure where this region should be hosted, but the existing property is working for the current implementation. We'll likely want to leave it as-is unless AWS has changed it to be the us-east-2

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just a couple comments, looking good.

File: src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java (hunk 6 above)
Comment (dmccoystephenson) [reply src/main/java/us/dot/its/jpo/ode/aws/depositor/AwsDepositor.java]: This change has been reverted.

File: .vscode/launch.json (hunk 1 above)
Comment (payneBrandon) [reply .vscode/launch.json:13]: If we're going to include the launch.json, let's document how to run it in the README. IE we're assuming the user has access to a kafka host as well as access/secret and appropriate keys. Is there a way to read those from a .env file maybe? I'm worried with having them forced to be replaced in this launch.json that someone will check in their configuration by accident and expose their keys.

**Review by payneBrandon — CHANGES_REQUESTED** [review]
Just a few outstanding issues.

File: .vscode/launch.json (hunk 1 above)
Comment (dmccoystephenson) [reply .vscode/launch.json:13]: I've added a section on launch configurations to the README.

That's fair. I think I initially tried to use environment variables, but I recall running into trouble. I'll do some testing to see if I can get it to work with environment variables.

Comment (dmccoystephenson) [reply .vscode/launch.json:13]: VSCode doesn't seem to let you set environment variables from environment variables (method shown below), and I don't think it's possible to load them in from a .env file.

```
"env": {
    "AWS_ACCESS_KEY_ID": "${TEST}",
    "AWS_SECRET_KEY": "${TEST}"
}
```

**Comment by dmccoystephenson** [comment]
This PR is ready for re-review @payneBrandon

**Review by payneBrandon — APPROVED** [review]

**Pull request merged**